include/revng/ADT/GenericGraph.h

```diff
@@ -15,14 +17,22 @@ struct Empty {};
 template<typename Node, size_t SmallSize = 16, bool HasEntryNode = true>
 class GenericGraph;
 
-template<typename T>
-class Parent {
+template<typename T, typename BaseType>
+class Parent : public BaseType {
+public:
+  template<typename... Args>
+  explicit Parent(Args &&... args) :
+    BaseType(std::forward<Args>(args)...), Par(nullptr) {}
+
+  Parent(const Parent &) = default;
+  Parent(Parent &&) = default;
+
 private:
-  T *Parent;
+  T *Par;
```

Comment:
Why? Abbreviations are not readable.

Reply:
when a costructor is added with the same name as the class, no field can have the same name as the calss

Reply:
What about `TheParent`?